src/synthesizer/blackholes.py

```diff
@@ -0,0 +1,460 @@
+                to be coarse. Defaults to None since it should be provided.
+            metallicity (float)
+                The metallicity of the NLR and BLR gas. Defaults to None since
+                it should be provided.
+            ionisation_parameter_blr (float)
+                The  ionisation parameter in the BLR. Default value
+                is 0.1.
+            hydrogen_density_blr (unyt.unit_object.Unit)
+                The  hydrogen density in the BLR. Default value
+                is 1E9/cm**3.
+            covering_fraction_blr (float)
+                The covering fraction of the BLR. Default value is 0.1.
+            velocity_dispersion_blr (unyt.unit_object.Unit)
+                The velocity disperson of the BLR. Default value is 
+                2000*km/s.
+            ionisation_parameter_nlr (float)
+                The ionisation parameter in the BLR. Default value
+                is 0.01.
+            hydrogen_density_nlr (unyt.unit_object.Unit)
+                The hydrogen density in the NLR. Default value
+                is 1E4/cm**3.
+            covering_fraction_nlr (float)
+                The covering fraction of the NLR. Default value is 0.1.
+            velocity_dispersion_nlr (unyt.unit_object.Unit)
+                The velocity disperson of the NLR. Default value is 
+                500*km/s.
+            theta_torus (float)
+                The opening angle of the torus component. Default value is
+                10*deg.
+            torus_emission_model (synthesizer.dust.emission)
+                The torus emission model. A synthesizer dust emission model.
+                Default is a Greybody with T=1000*K and emissivity=1.6.
+            **kwargs 
+                Addition parameters for the specific disc model.
+
+        """
+
+        # Create a dictionary of all the arguments (and self)
+        args = locals()
+        
+        # Save model and directory as attributes.
+        self.disc_model = disc_model
+        self.photoionsation_model = photoionisation_model
+        self.grid_dir = grid_dir
+
+        # Save the torus model
+        self.torus_emission_model = torus_emission_model
+
+        # These are the unified model parameters, i.e. all the non-disc and
+        # non-torus parameters that are needed.
+        self.unified_parameters = [
+            'bolometric_luminosity',
+            'metallicity',
+            'ionisation_parameter_blr',
+            'hydrogen_density_blr',
+            'covering_fraction_blr',
+            'velocity_dispersion_blr',
+            'ionisation_parameter_nlr',
+            'hydrogen_density_nlr',
+            'covering_fraction_nlr',
+            'velocity_dispersion_nlr',
+            'theta_torus',
+        ]
+
+        # Create dictionary of fixed parameters based on what was handed
+        self.fixed_parameters_dict = {}
+        for parameter in self.unified_parameters:
+            if args[parameter]:
```

Comment:
This should be `if args[parameter] is not None` to remove ambiguity. Although `args` should probably be switched up to remove `locals`: see comment above

Reply:
Done